pb_encode.c

```diff
@@ -626,14 +626,19 @@ bool checkreturn pb_encode_submessage(pb_ostream_t *stream, const pb_field_t fie
 #endif
         return false;
     }
-    
+
     size = substream.bytes_written;
     if(size > VARINT_ONE_BYTE)
     {
         if (!pb_shift_data_1_byte(stream, substream.bytes_written))
             return false;
+        substream.state = (pb_byte_t*)substream.state + 1;
```

Comment:
can you check if this will cause an issue for fixed size fields?

Reply:
How do you mean? 
Fixed size fields use a different encoding function - this section of code is only executed during the encoding of a submessage (always defined as variable size).